### PR TITLE
Bl-30:tamhom - Implement and update for print out an interesting squa…

### PR DESCRIPTION
**Pull request by tamhom**
Exercise #30

## Overview

Implement and update for print out an interesting square with 0, + and -

## Verification Steps

This exercise is for practicing for loop and if statements.

Write the program that allows a user to input N, then print the square as below:

Please input N:10
```
O  O  O  O  O  O  O  O  O  O  
O  O  -  -  -  -  -  -  -  O 

O  +  O  -  -  -  -  -  -  O 

O  +  +  O  -  -  -  -  -  O 

O  +  +  +  O  -  -  -  -  O 

O  +  +  +  +  O  -  -  -  O 

O  +  +  +  +  +  O  -  -  O 

O  +  +  +  +  +  +  O  -  O 

O  +  +  +  +  +  +  +  O  O 

O  O  O  O  O  O  O  O  O  O 
```

### CODE REVIEW

File: python_basic/tamhom/print_square.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    for j in range(n):
+        if i == 0 or j == 0 or i == (n-1) or j == (n-1) or i == j:
+            sys.stdout.write('*\t')
+        else:
+            sys.stdout.write(' ')            
+            if j > i:
+                sys.stdout.write('-\t')
+            else:
+                sys.stdout.write('+\t')
+    print ''
```
Comment (hungphamviet) [inline python_basic/tamhom/print_square.py]: Why you need to add tabs after *, - and +? It might print a square not the same as in requirement.

**Comment by tamhom** [comment]
Thank for your review anh @hungphamviet, I've changed my codes and commit again. Please help to check when you have free time.

File: exercises/tamhom/print_square.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+__author__ = 'tamhom'
+
+import sys
+n = input("Enter n: ")
+
+for i in (range(n)):
```
Comment (hungphamviet) [inline exercises/tamhom/print_square.py]: No need parentheses around range(n)

File: exercises/tamhom/print_square.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    for j in range(n):
+        if i == 0 or j == 0 or i == (n-1) or j == (n-1) or i == j:
```
Comment (hungphamviet) [inline exercises/tamhom/print_square.py]: In Python, there are no need to have parentheses to border condition. Please simplify as `if i == 0 or j == 0 or i == n-1 or j == n-1 or i == j:`

File: exercises/tamhom/print_square.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+            sys.stdout.write('*')
```
Comment (hungphamviet) [inline exercises/tamhom/print_square.py]: To simplify, please change `sys.stdout.write` to `print`. See [this example](https://stackoverflow.com/questions/11266068/python-avoid-new-line-with-print-command)

File: exercises/tamhom/print_square.py (hunk 3 above)
Comment (tamhom) [inline exercises/tamhom/print_square.py]: I've removed parentheses around rang(n) and the condition. Also changed the code from ```sys``` to ```print```. Please review again.

Thanks,
Tam

**Comment by hungphamviet** [comment]
Merged #60 into `develop`

**Pull request merged**